### PR TITLE
config, charset: make charset config not affected by collation config

### PR DESCRIPTION
**Pull request by xiongjiwei**
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -2437,9 +2437,9 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
 
 	if newCollationEnabled {
 		collate.EnableNewCollations()
-		if cfg.Experimental.EnableNewCharset {
-			collate.EnableNewCharset()
-		}
+	}
+	if cfg.Experimental.EnableNewCharset {
```
Comment (tangenta) [inline session/session.go:2441]: So the default collation of GBK is still `gbk_chinese_ci`?

Comment (xiongjiwei) [reply session/session.go:2441]: yes, but it will not be respected.

Comment (zimulala) [reply session/session.go:2441]: Shall we deal with it? Feels like a good idea to ban it. Besides, we can add related tests for `gbk_bin` and `gbk_chinese_ci`.

Comment (xiongjiwei) [reply session/session.go:2441]: no, there is nothing should do, just like the collation of the other charsets. if the new collation is not enabled, we can use any collation but the result does not respect the used collation.

Comment (zimulala) [reply session/session.go:2441]: Get it. Just add two tests? For example, get the results of `show collation` when `EnableNewCharset` is true and `newCollationEnabled` is false.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**